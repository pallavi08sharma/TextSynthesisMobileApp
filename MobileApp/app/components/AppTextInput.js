import React from "react";
import { View, TextInput, StyleSheet } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";

import defaultStyles from "../config/styles";
import colors from "../config/colors";

function AppTextInput({ icon, ...otherProps }) {
  return (
    <View style={styles.textInputcontainer}>
      {icon && (
        <MaterialCommunityIcons
          name={icon}
          size={20}
          color={defaultStyles.colors.white}
          style={styles.icon}
        />
      )}
      <TextInput style={defaultStyles.text} {...otherProps} />
    </View>
  );
}

const styles = StyleSheet.create({
  textInputcontainer: {
    backgroundColor: colors.greymedium,
    borderRadius: 25,
    flexDirection: "row",
    width: "100%",
    padding: 10,
    paddingLeft: 15,
    marginVertical: 10,
    //flexWrap: 'wrap'
  },
  icon: {
    marginRight: 10,
  },
});

export default AppTextInput;
