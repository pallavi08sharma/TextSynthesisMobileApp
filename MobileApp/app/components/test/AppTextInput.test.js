import React from 'react';
import renderer from 'react-test-renderer';

import AppTextInput from '../AppTextInput';

describe('<AppTextInput />', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<AppTextInput />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });