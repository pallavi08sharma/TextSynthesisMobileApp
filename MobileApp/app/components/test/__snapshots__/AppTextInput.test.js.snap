// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppTextInput /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#d3d3d3",
      "borderRadius": 25,
      "flexDirection": "row",
      "marginVertical": 10,
      "padding": 10,
      "paddingLeft": 15,
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#0c0c0c",
        "fontFamily": "Avenir",
        "fontSize": 18,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
